src/ja/worker/config.py

```diff
@@ -1,36 +1,76 @@
 from typing import Dict
-
 from ja.common.config import Config
 from ja.common.proxy.ssh import SSHConfig
 from ja.common.work_machine import ResourceAllocation
+import yaml
+import os
 
 
 class WorkerConfig(Config):
     """
     Config for the worker client.
     """
 
+    def __init__(self, conf_path: str):
+        """!
+        reads the config file from the disk
+        creates resource allocation instance
+        """
+        self._conf_path = conf_path
+        self._ssh_config = self.read_conf()
+        self._resource_allocation = self.get_resources()
+
+    def read_conf(self) -> SSHConfig:
```

Comment:
This method is not needed.
YAML files can be automatically converted into Python dictionaries.
We can therefore just use the from_dict method to convert a YAML file to a Python object.